src/plasma/plasma_manager.cc

```diff
@@ -674,6 +687,9 @@ void process_data_chunk(event_loop *loop,
   int err = read_object_chunk(conn, buf);
   auto plasma_conn = conn->manager_state->plasma_conn;
   if (err != 0) {
+    // Remove the object from the receives_in_progress set so that
+    // retries are processed.
+    conn->manager_state->receives_in_progress.erase(buf->object_id);
```

Comment:
Could you add an additional comment here with a TODO about flushing the whole thing if we allow parallel object transfers? I saw the comment above, but it would be good to include here as well.